### PR TITLE
GUVNOR-2851, GUVNOR-2870, GUVNOR-2835: Library perspective

### PR DESCRIPTION
**Pull request by paulovmr**
* Moved 'LibraryMonitor' to another project.

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:

https://github.com/droolsjbpm/kie-wb-common/pull/655
https://github.com/droolsjbpm/drools-wb/pull/393
https://github.com/droolsjbpm/jbpm-form-modeler/pull/86
https://github.com/droolsjbpm/jbpm-designer/pull/543
https://github.com/droolsjbpm/kie-wb-distributions/pull/453

File: kie-drools-wb/kie-drools-wb-home-page-community/src/main/java/org/kie/workbench/drools/client/home/HomeProducer.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import org.kie.workbench.common.screens.home.model.HomeModel;
 import org.kie.workbench.common.screens.home.model.ModelUtils;
 import org.kie.workbench.common.screens.home.model.SectionEntry;
-import org.kie.workbench.common.screens.library.client.monitor.LibraryMonitor;
+import org.kie.workbench.common.workbench.client.library.LibraryMonitor;
```
Comment (psiroky) [inline kie-drools-wb/kie-drools-wb-home-page-community/src/main/java/org/kie/workbench/drools/client/home/HomeProducer.java]: Maybe it's just me, but `org.kie.workbench.common.workbench` sounds weird :)

Comment (paulovmr) [inline kie-drools-wb/kie-drools-wb-home-page-community/src/main/java/org/kie/workbench/drools/client/home/HomeProducer.java]: @psiroky Indeed it is redundant... Best to leave the change to another PR, since the package already existed and is used by all webapps. I opened a [JIRA issue](https://issues.jboss.org/browse/GUVNOR-2871) for that.

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

File: kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectAuthoringPerspective.java [hunk 2]
```diff
@@ -71,6 +76,10 @@ public ProjectExplorer getProjectExplorer() {
     }
 
     public ProjectEditor openProjectEditor() {
+        if ( !Waits.isElementPresent( PROJECT_EXPLORER_TITLE ) ) {
```
Comment (manstis) [inline kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectAuthoringPerspective.java]: Yay!

File: kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/ProjectAuthoringIntegrationTest.java [hunk 3]
```diff
@@ -33,19 +35,24 @@
 
     private static final Logger LOG = LoggerFactory.getLogger( ProjectAuthoringIntegrationTest.class );
 
+    private HomePerspective home;
+
+    @Before
+    public void setup() {
```
Comment (jhrcek) [inline kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/ProjectAuthoringIntegrationTest.java]: Does it make sense to import mortgage project before every test in this class? The common mistake people do when writing selenium tests is writing them as if they were unit tests. From QE's experience every 10 seconds added to the test execution time makes the tests less valuable and people more likely to skip wasting time on their maintenance ;-)

File: kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectLibraryPerspective.java [hunk 4]
```diff
@@ -15,53 +15,35 @@
...
 import org.kie.wb.selenium.util.Waits;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 
 public class ProjectLibraryPerspective extends AbstractPerspective {
 
-    //<div class="full" data-field="emptyLibrary">
-    private static final By PROJECT_LIBRARY_HOLDER = By.xpath( "//div[@data-field='emptyLibrary']" );
+    private static final String OU_REPO_BREADCRUMB_XPATH_SELECTOR = "//a[text()='myteam (myrepo)']";
```
Comment (jhrcek) [inline kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectLibraryPerspective.java]: I'd recommend replacing this by more readable `By.linkText("myteam (myrepo)")`

Also you can use `@FindByJquery` annotation or `ByJQuery` from Graphene to create locators that are more readable.

e.g. replace
```
private static final String MORTGAGES_BUTTON_XPATH_SELECTOR = "//button[text()='mortgages']";
private static final By MORTGAGES_IMPORT_BUTTON = By.xpath(MORTGAGES_BUTTON_XPATH_SELECTOR);
@FindBy(xpath = MORTGAGES_BUTTON_XPATH_SELECTOR)
private WebElement exampleButton;
```
By more readable
```
@FindByJquery("button:contains(mortgages)")
private WebElement exampleButton;
```

**Comment by karreiro** [comment]
Jenkins, please retest this.

**Comment by csadilek** [comment]
jenkins retest this.

**Comment by csadilek** [comment]
There's a network issue it seems...

**Comment by paulovmr** [comment]
jenkins retest this

File: kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/ProjectAuthoringIntegrationTest.java (hunk 3 above)
Comment (paulovmr) [inline kie-wb-tests/kie-wb-tests-gui/src/test/java/org/kie/wb/selenium/ui/ProjectAuthoringIntegrationTest.java]: @jhrcek Hi, since it's a test for Authoring and the only way to access authoring is if some accessible project already exists in the workbench, I think it's fair that the `setup` method guarantees that there is a project and the tests can access Authoring. :)

Also, @bsremac will work on this tests, maybe we can have some improvements in that aspect. :)

File: kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectLibraryPerspective.java (hunk 4 above)
Comment (paulovmr) [inline kie-wb-tests/kie-wb-tests-gui/src/main/java/org/kie/wb/selenium/model/persps/ProjectLibraryPerspective.java]: @jhrcek I agree, we could explore more the `@FindByJQuery` annotation. 

@bsremac After this PR get merged, maybe it's a good thing to improve. :)

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
@csadilek @manstis Finally green! :D
